### ISSUE TITLE
SelectElement: various getText()-methods are not implemented properly

### ISSUE THREAD

**Issue by Slava569845** [vaadin/testbench#1686]
Hello, we're using Vaadin 23 and doing selenium-tests quite a lot. We recently found, that extracting a simple text from a simple Select-item is quite cumbersome with provided SelectElement. I have build a small test case (maven project, attached) demonstrating the issue:

Production view:
`@Route("demo")
public class SelectDemo extends VerticalLayout
{
    public SelectDemo()
    {
        Select<String> select = new Select<>();
        select.setItems("1", "2", "3");
        select.setValue("2");
        add(select);
    }
}`

Selenium test:
`@Test
public void test()
{
    driver.navigate().to("http://localhost:8090/demo");
    SelectElement selectElement = $(SelectElement.class).waitForFirst();
    LOGGER.info("Items.getText: {}", selectElement.getItems().stream().map(SelectElement.ItemElement::getText).collect(Collectors.toList()));
    LOGGER.info("Items.getPropertyString(textContent): {}", selectElement.getItems().stream().map(i -> i.getPropertyString("textContent")).collect(Collectors.toList()));
    LOGGER.info("getSelectedItem().getText(): {}", selectElement.getSelectedItem().getText());
    LOGGER.info("getSelectedText(): {}", selectElement.getSelectedText());
}`

Output:
[main] INFO de.akdb.SelectDemoSeleniumTest - Items.getText: [, , ]
[main] INFO de.akdb.SelectDemoSeleniumTest - Items.getPropertyString(textContent): [1, 2, 3]
[main] INFO de.akdb.SelectDemoSeleniumTest - getSelectedItem().getText(): 
[main] INFO de.akdb.SelectDemoSeleniumTest - getSelectedText(): 

Basically various getText methods are not working as one would intuitively expect them to and we have to resort to dubios workaround getPropertyString("textContent"), which may depend on the implementation and thus Vaadin version i'm afraid.

**Comment by Slava569845** [comment]
[selectelement.zip](https://github.com/vaadin/testbench/files/12879873/selectelement.zip)
